pkg/controllers/repositoryserver/handler.go

```diff
@@ -0,0 +1,300 @@
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/poll"
+)
+
+type RepoServerHandler struct {
+	Req                     ctrl.Request
+	Logger                  logr.Logger
+	Reconciler              *RepositoryServerReconciler
+	KubeCli                 kubernetes.Interface
+	RepositoryServer        *crv1alpha1.RepositoryServer
+	RepositoryServerSecrets repositoryServerSecrets
+}
+
+func (h *RepoServerHandler) CreateOrUpdateOwnedResources(ctx context.Context) error {
+	svc, err := h.reconcileService(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to reconcile service")
+	}
+	if err = h.getSecretsFromCR(ctx); err != nil {
+		return errors.Wrap(err, "failed to get repository server secrets")
+	}
+	pod, err := h.reconcilePod(ctx, svc)
+	if err != nil {
+		return errors.Wrap(err, "failed to reconcile repository server pod")
+	}
+	if err := h.waitForPodReady(ctx, pod); err != nil {
+		return errors.Wrap(err, "repository server pod not in ready state")
+	}
+	if err := h.connectToKopiaRepository(); err != nil {
+		return errors.Wrap(err, "failed to connect to kopia repository")
+	}
+
+	if err := h.startRepoProxyServer(ctx); err != nil {
+		return errors.Wrap(err, "failed to start kopia repository servver")
+	}
+
+	if err := h.createOrUpdateClientUsers(ctx); err != nil {
+		return errors.Wrap(err, "failed to create/update kopia server access users")
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) reconcileService(ctx context.Context) (*corev1.Service, error) {
+	repoServerNamespace := h.RepositoryServer.Namespace
+	serviceName := h.RepositoryServer.Status.ServerInfo.ServiceName
+	svc := &corev1.Service{}
+	h.Logger.Info("Check if Service resource exists. If exists, reconcile with CR spec")
+	err := h.Reconciler.Get(ctx, types.NamespacedName{Name: serviceName, Namespace: repoServerNamespace}, svc)
+	if err == nil {
+		return h.updateService(ctx, svc)
+	}
+	if !apierrors.IsNotFound(err) {
+		return nil, err
+	}
+	h.Logger.Info("Service resource not found. Creating new service")
+	svc, err = h.createService(ctx, repoServerNamespace)
+	if err != nil {
+		return nil, err
+	}
+	h.Logger.Info("Update serviceName in RepositoryServer /status")
+	serverInfo := crv1alpha1.ServerInfo{
+		PodName:     h.RepositoryServer.Status.ServerInfo.PodName,
+		ServiceName: svc.Name,
+	}
+	if err := h.updateServerInfoInCRStatus(ctx, serverInfo); err != nil {
+		return nil, errors.Wrap(err, "Failed to update ServiceName in RepositoryServer /status")
+	}
+	return svc, err
+}
+
+func (h *RepoServerHandler) updateService(ctx context.Context, svc *corev1.Service) (*corev1.Service, error) {
+	svc = h.updateServiceSpecInService(svc)
+	if err := h.Reconciler.Update(ctx, svc); err != nil {
+		return nil, err
+	}
+	return svc, nil
+}
+
+func (h *RepoServerHandler) updateServiceSpecInService(svc *corev1.Service) *corev1.Service {
```

Comment:
Do we need `InService` in the name `updateServiceSpecInService`? Is it not enough if we say `updateServiceSpec`?